plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -82,6 +83,37 @@ public function __construct() {
 		}
 	}
 
+
+	/**
+	 * Adds the product template ID to the product if it doesn't exist.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WC_Product       $product The product.
+	 */
+	public function possibly_add_template_id( $response, $product ) {
+		if ( ! $product ) {
+			return $response;
+		}
+		if ( ! $product->meta_exists( '_product_template_id' ) ) {
+			/**
+			 * Experimental: Allows to determine a product template id based on the product data.
+			 *
+			 * @ignore
+			 * @since 9.1.0
+			 */
+			$product_template_id = apply_filters( 'experimental_woocommerce_product_editor_product_template_for_product', '', $product );
```

Comment:
Hmmm tagging `experimental_` at the front of a hook is that the expected way to label it as experimental?
I would lean towards removing that, as this does mean we have to deprecate it when it becomes official. But if this is normal across WP than I did be ok with this.
Also we are looking for a product template `id` here, so I think we should change the latter part to add `_id` in -> `product_template_id_for_product`.